deploy.md

```diff
@@ -1,59 +1,149 @@
+
+  Important: you need to update the branch name in the `deploy` section in [travis.yml](.travis.yml) to let CI build the release candiate every once you push a new commit to the `release-X.Y.Z` branch.
+
+  As a side note, there are three channels for Activity Stream.
+  * [dev](https://s3.amazonaws.com/moz-activity-streams-dev/dist/latest.html)
+  * [pre-release](https://s3.amazonaws.com/moz-activity-streams-prerelease/dist/latest.html)
+  * [release](https://moz-activity-streams.s3.amazonaws.com/dist/latest.html)
 
-`npm install`
+- Release signing-off. Upon the first build of release candidate, the release engineer sends out the release preview email to all the stakeholders. In addition, she/he makes a release pull request on GitHub to track the status through an attached checklist. The release deploy won't be initiated unless all the items get checked by the stakeholders.
 
-# Update version numbers
-- set a new version of the addon by running npm version
+  Over the course of this stage, QA tests the pre-release build and files bugs with the label "release/block". Meanwhile, PM and UI/UX verifies the add-on and gives feedback. Engineers, in turn, respond to those inputs and make bugfixes or changes accordingly.
 
-`npm version patch`
+- Release deploy. As the final stage of the release process, the release engineer ensures all requirements are met on the checklist, then proceeds to deploy the release as described below in detail.
 
-- update the changelog:
+Note that the first three tasks are mostly driven by the automated "pre-release" channel, people from different release party could cooperate directly on GitHub or via IRC/Slack.
 
-`./node_modules/.bin/conventional-changelog -i CHANGELOG.md -s -r 0`
+Let's now focus on the final task - release deploy. Again, we break it down into three stages as pre-deploy, deploy, and post-deploy respectiviely.
 
-`git add CHANGELOG.md`
+# Pre-deploy
 
-`git commit -m '1.0.X changelog' # where X is the patch version you just upgraded to`
+In this stage, we prep everything for the release in the git repo.
 
-- update remote repository
+## Update code and packages
 
-`git push origin master`
+- ensure you have the latest version of the code
+
+  ```
+  git checkout release-X.Y.Z
+  git pull upstream release-X.Y.Z
+  ```
+
+- update/install any packages
 
-`git push origin --tags`
+  ```
+  npm install
+  ```
+
+## Update CHANGELOG and version numbers
+- set a new version of the add-on by running npm version
+
+  ```
+  npm version patch
+  ```
+
+- update the changelog
+
+  ```
+  npm run changelog -- -r 0
+  git add CHANGELOG.md
+  git commit -m 'X.Y.Z changelog' # where X, Y, and Z are the major, minor, and patch version you just upgraded to
+  ```
+
+## Update repository
+- merge the latest changelog to master
+  ```
+  # merge changes to master
+  git checkout master
+  git merge release-X.Y.Z
```

Comment:
We can't merge everything in the release branch into master.  This should be more explicit about merging only the changelog changes back in, not all commits in the release branch.